### PR TITLE
feat(cart): remove coupon

### PR DESCRIPTION
**Pull request by nsilva-spryker**
- add remove button to delete coupon in coupon component
- update services and adapters to remove coupon and update cart totals
- update component unit and e2e tests coverage

closes: [HRZ-90814](https://spryker.atlassian.net/browse/HRZ-90814)

[HRZ-90814]: https://spryker.atlassian.net/browse/HRZ-90814?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/cart/coupon/coupon.component.ts [hunk 1]
```diff
@@ -67,7 +67,17 @@ export class CouponComponent extends CartComponentMixin(
           return html`
             <div>
               <oryx-icon .type="${IconTypes.Check}"></oryx-icon>
-              <span class="code">${coupon.code} </span>
+              <span class="code">
+                ${coupon.code}
+                <oryx-button
+                  .type=${ButtonType.Icon}
```
Comment (Sinucid) [inline libs/domain/cart/coupon/coupon.component.ts:74]: need to add aria-label, the button without text inside does not pass accessibility check

File: libs/domain/cart/services/src/adapter/default-cart.adapter.ts [hunk 2]
```diff
@@ -163,6 +163,22 @@ export class DefaultCartAdapter implements CartAdapter {
...
+      take(1),
+      switchMap((identity) => {
+        const url = this.generateUrl(
+          identity.isAuthenticated
+            ? `${ApiCartModel.UrlParts.Carts}/${data.cartId}/${ApiCartModel.UrlParts.Coupons}/${data.code}`
+            : `${ApiCartModel.UrlParts.GuestCarts}/${data.cartId}/${ApiCartModel.UrlParts.Coupons}/${data.code}`,
+          !identity.isAuthenticated
+        );
+
+        return this.http.delete(url);
```
Comment (Sinucid) [inline libs/domain/cart/services/src/adapter/default-cart.adapter.ts]:
```suggestion
        const requestType =  identity.isAuthenticated ? 
        ApiCartModel.UrlParts.Carts: 
        ApiCartModel.UrlParts.GuestCarts;
        
        const url = this.generateUrl(          
          `${requestType}/${data.cartId}/${ApiCartModel.UrlParts.Coupons}/${data.code}`,
          !identity.isAuthenticated
        );

        return this.http.delete(url);
```

File: libs/domain/cart/services/src/default-cart.service.ts [hunk 3]
```diff
@@ -101,12 +102,20 @@ export class DefaultCartService implements CartService {
     },
   });
 
+  protected removeCouponCommand$ = createCommand({
+    ...this.cartCommandBase,
+    action: (qualifier: CouponQualifier) => {
+      return this.adapter.deleteCoupon(qualifier);
+    },
+    onSuccess: [...this.cartCommandBase.onSuccess, CartEntryRemoved],
```
Comment (Sinucid) [inline libs/domain/cart/services/src/default-cart.service.ts]: Probably query events are mixed up, this uses `CartEntryRemoved`, and entry command - `CouponRemoved`

File: libs/domain/cart/services/src/default-cart.service.ts [hunk 4]
```diff
@@ -287,6 +296,10 @@ export class DefaultCartService implements CartService {
     return this.executeWithOptionalCart(qualifier, this.addCouponCommand$);
   }
 
+  deleteCoupon(qualifier: CouponQualifier): Observable<unknown> {
```
Comment (Sinucid) [inline libs/domain/cart/services/src/default-cart.service.ts:299]: and need to add this method to the cart mock

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/domain/cart/services/src/default-cart.service.ts (hunk 4 above)
Comment (nsilva-spryker) [reply libs/domain/cart/services/src/default-cart.service.ts:299]: Coupon service will be set in different service and adapter in a follow up story, pls check: https://spryker.slack.com/archives/C029XPSBXC6/p1704820725966759

**Review by Sinucid — APPROVED** [review]

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**